zenoh-kuksa-provider/src/main.rs

```diff
@@ -39,69 +42,59 @@ struct Args {
     config: String,
 }
 
-fn read_config(path: &str) -> Result<ProviderConfig, Box<dyn Error>> {
-    let config_str = fs::read_to_string(path)?;
-    let config = json5::from_str(&config_str)?;
+impl Args {
+    fn read_config(&self) -> Result<ProviderConfig, Box<dyn std::error::Error>> {
+        let config_str = fs::read_to_string(&self.config)?;
+        let config = json5::from_str(&config_str)?;
 
-    Ok(config)
+        Ok(config)
+    }
 }
 
-async fn handling_zenoh_subscribtion(
-    provider_config: Arc<ProviderConfig>,
+async fn handling_zenoh_subscription(
     session: Arc<Session>,
     metadata_store: MetadataStore,
     mut kuksa_client: kuksa::Client,
 ) {
-    info!("Listening on selector: {:?}", provider_config.zenoh.key_exp);
+    info!("Listening on selector: {:?}", VEHCILE_KEY_EXPR);
 
-    let provider_config_clone = Arc::clone(&provider_config);
-    let subscriber = session
-        .declare_subscriber(provider_config_clone.zenoh.key_exp.clone())
-        .res()
-        .await
-        .unwrap();
+    let subscriber = session.declare_subscriber(VEHCILE_KEY_EXPR).await.unwrap();
 
     let store = metadata_store.lock().await;
 
     while let Ok(sample) = subscriber.recv_async().await {
-        let vss_path = sample.key_expr.replace("/", ".").to_string();
+        let vss_path = sample.key_expr().replace("/", ".").to_string();
 
         debug!(
-            "Received ('{}': '{}' with attachment: {:?})",
-            &vss_path, sample.value, sample.attachment
+            "Received ('{}': '{:?}' with attachment: {:?})",
+            &vss_path,
+            sample.payload(),
+            sample.attachment()
         );
 
-        let field_type = extract_attachment_as_string(&sample);
-
-        if field_type == "currentValue" {
-            let datapoint_update = new_datapoint_for_update(&vss_path, &sample, &store);
-
-            debug!("Forwarding: {:#?}", datapoint_update);
-            kuksa_client
-                .set_current_values(datapoint_update)
-                .await
-                .unwrap();
+        if let Some(field_type) = extract_attachment_as_string(&sample) {
```

Comment:
no else case? intended?

Reply:
there is nothing to do if the sample has no attachment (i.e. data)